patch/llpcPatchInOutImportExport.cpp

```diff
@@ -3152,6 +3159,48 @@ Value* PatchInOutImportExport::PatchFsBuiltInInputImport(
                                       pInsertPos);
             break;
         }
+    case BuiltInSamplePosOffset:
+        {
+            // Gets the offset of sample position relative to the pixel center for the specified sample ID
+            IRBuilder<> builder(*m_pContext);
+            builder.SetInsertPoint(pInsertPos);
+            Value* pNumSamples = PatchFsBuiltInInputImport(builder.getInt32Ty(), BuiltInNumSamples, pInsertPos);
+            Value* pPatternIdx = PatchFsBuiltInInputImport(builder.getInt32Ty(), BuiltInSamplePatternIdx, pInsertPos);
+            Value* pAdd = builder.CreateAdd(pPatternIdx, pSampleId);
+            // offset = (sampleCount > sampleId) ? (samplePatternOffset + sampleId) : 0
+            Value* pNotTooBig = builder.CreateICmpUGT(pNumSamples, pSampleId);
+            Value* pOffset = builder.CreateSelect(pNotTooBig, pAdd, builder.getInt32(0));
+            // Load sample position descriptor.
+            auto pDesc = EmitCall(m_pModule,
+                                  LlpcName::DescriptorLoadBuffer,
+                                  VectorType::get(builder.getInt32Ty(), 4),
+                                  {
+                                      builder.getInt32(InternalResourceTable),
+                                      builder.getInt32(SI_DRV_TABLE_SAMPLEPOS),
+                                      builder.getInt32(0),
+                                  },
+                                  NoAttrib,
+                                  pInsertPos);
+            pOffset = builder.CreateShl(pOffset, builder.getInt32(4));
+            pInput = builder.CreateIntrinsic(Intrinsic::amdgcn_raw_buffer_load,
+                                             pInputTy,
+                                             {
+                                                pDesc,
+                                                pOffset,
+                                                builder.getInt32(0),
+                                                builder.getInt32(0)
+                                             });
+            break;
+        }
+    case BuiltInSamplePosition:
+        {
+            IRBuilder<> builder(*m_pContext);
```

Comment:
Avoid use IR builder to keep codes consistent.

Reply:
As above.